### ISSUE TITLE
XMLHttpRequest's using Git WebView Resource return unexpected result

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/vscode#115614]
- VSCode Version:  Version: 1.52.1 (user setup)
- OS Version: Windows 10, also Linux

TLDR: `XMLHttpRequest` inside a WebView on a Git link like
```
vscode-webview-resource://ba65fa50-10a1-41ed-81ee-f4425d2254f9/git/d%3A/pdf/dummy.pdf?%7B%22path%22%3A%22d%3A%5C%5Cpdf%5C%5Cdummy.pdf%22%2C%22ref%22%3A%22~%22%7D 
```
does not provide a valid pdf file, but plain text from `git diff` instead.

Steps to Reproduce:

1. Prepare folder
```
mkdir D:\pdf
cd /d D:\pdf
git init
curl https://www.w3.org/WAI/ER/tests/xhtml/testfiles/resources/pdf/dummy.pdf --output dummy.pdf
git add dummy.pdf
git commit -m "dummy.pdf"
curl http://www.africau.edu/images/default/sample.pdf --output dummy.pdf
git diff dummy.pdf
```

You should see something like this:
```
D:\pdf>git diff dummy.pdf
warning: CRLF will be replaced by LF in dummy.pdf.
The file will have its original line endings in your working directory
diff --git a/dummy.pdf b/dummy.pdf
index 774c2ea..c0e31a0 100644
--- a/dummy.pdf
+++ b/dummy.pdf
@@ -1,2 +1,22 @@
-Dummy PDF file
+A Simple PDF File
+
+  This is a small demonstration .pdf file -
+
+  just for use in the Virtual Mechanics tutorials. More text. And more
+  text. And more text. And more text. And more text.
```

2. `code D:\pdf`
3. Install https://marketplace.visualstudio.com/itemdetails?itemName=tomoki1207.pdf (instructions below are for v1.1.0)
4. Go to source control view
5. Double-click `dummy.pdf`

You will now see this - this is the bug described in https://github.com/tomoki1207/vscode-pdfviewer/issues/70
![image](https://user-images.githubusercontent.com/12128514/106616738-29fa2780-656e-11eb-9728-e9975c77613b.png)

To see what the problem is:
6. Open `%USERPROFILE%\.vscode\extensions\tomoki1207.pdf-1.1.0\lib\build\pdf.js` (`$HOME/.vscode-server/extensions/tomoki1207.pdf-1.1.0/lib/build/pdf.js` for Remote SSH server on Linux)
7. Add `console.log(this.url, xhr);` before line 12738 (`const chunk = getArrayBuffer(xhr);`); save and reload as needed
8. Go to source control view again
9. Double-click `dummy.pdf`
10. Start `workbench.action.webview.openDeveloperTools` action
11. Expand `XMLHttpRequest` in the console windows that just opened.

You will see two requests like this:

First:
![image](https://user-images.githubusercontent.com/12128514/106617117-9d9c3480-656e-11eb-889d-d8dff756e620.png)

Second:
![image](https://user-images.githubusercontent.com/12128514/106617170-aa208d00-656e-11eb-91ca-5b0845e078ac.png)

Note the `response` for the second being only 16 bytes long, while the first is >3000 bytes. The first (`37, 80, 68, 70, 45, 49, 46, 51, 13, 10, 37 ...`) is basically `%PDF-1.3...`, a valid PDF header - while the second (`68, 117, 109, 109, 121, 32, 80, 68, 70, 32, 102, 105, 108, 101, 10, 12`) is `Dummy PDF file`. This is exactly what `git diff` shows above. I feel the WebView resource resolver should provide the full `pdf` file, not just its text.

**Comment by bersbersbers** [comment]
I should add that the behavior on Windows with Remote SSH to Linux is a bit different. In this case, both requests return the new pdf file, which is also confirmed by the requests having the same size (3028 bytes):

![image](https://user-images.githubusercontent.com/12128514/106632358-f32c0d80-657d-11eb-8476-af843d39e59a.png)
![image](https://user-images.githubusercontent.com/12128514/106632596-3a1a0300-657e-11eb-8e5d-1cfede9fb65c.png)
![image](https://user-images.githubusercontent.com/12128514/106632605-3c7c5d00-657e-11eb-9fd7-0fe931e3f055.png)

**Comment by bersbersbers** [comment]
@eamodio I don't agree that this issue is *caused* by an extension - yes, it is observed in an extension, but the root cause is in incorrectly(?) resolving the Git WebView resource.

The resolving of the resource is in line with the command-line output of `git diff`. On a clean installation of `git` on Linux, `git diff`, one gets
```
diff --git a/dummy.pdf b/dummy.pdf
index 774c2ea..dbf091d 100644
Binary files a/dummy.pdf and b/dummy.pdf differ
lines 1-3/3 (END)
```
and then the issue does *not* appear.

The issue *does* appear, however, when one uses (still on Linux) a `.gitattributes` files with 
```
*.pdf           diff=pdf
```
and a `.gitconfig` with
```
[diff "pdf"]
        textconv = pdfinfo
        binary = true
```
In this case, `git diff` shows something like this:
```
diff --git a/dummy.pdf b/dummy.pdf
index 774c2ea..dbf091d 100644
--- a/dummy.pdf
+++ b/dummy.pdf
@@ -1,16 +1,15 @@
-Author:         Evangelos Vlachogiannis
-Creator:        Writer
-Producer:       OpenOffice.org 2.1
-CreationDate:   Fri Feb 23 16:56:37 2007 CET
+Creator:        Rave (http://www.nevrona.com/rave)
+Producer:       Nevrona Designs
+CreationDate:   Wed Mar  1 08:28:26 2006 CET
 Tagged:         no
 UserProperties: no
```
and the same seems to be true in resolving the WebView resource, leading to apparently broken PDF files.

On Windows (using https://gitforwindows.org/), I don't recall making any of these changes, and I cannot find a `.gitattributes` file. However, a default setting `diff.astextplain.textconv=astextplain` likely has the same effect (compare https://stackoverflow.com/q/28146244/).

```
D:\pdf>git config --list --show-origin
file:C:/Program Files/Git/etc/gitconfig http.sslbackend=schannel
file:C:/Program Files/Git/etc/gitconfig diff.astextplain.textconv=astextplain
file:C:/Program Files/Git/etc/gitconfig credential.helper=manager-core
file:C:/Program Files/Git/etc/gitconfig core.editor="C:\\Program Files\\Notepad++\\notepad++.exe" -multiInst -notabbar -nosession -noPlugin
file:C:/Program Files/Git/etc/gitconfig core.autocrlf=input
file:C:/Program Files/Git/etc/gitconfig core.fscache=true
file:C:/Program Files/Git/etc/gitconfig core.symlinks=false
file:C:/Program Files/Git/etc/gitconfig pull.rebase=false
file:C:/Program Files/Git/etc/gitconfig credential.https://dev.azure.com.usehttppath=true
file:C:/Program Files/Git/etc/gitconfig init.defaultbranch=master
file:C:/Users/bers/.gitconfig   color.diff.old=white red
file:C:/Users/bers/.gitconfig   color.diff.new=white green
file:C:/Users/bers/.gitconfig   winupdater.recentlyseenversion=2.25.0.windows.1
file:.git/config        core.repositoryformatversion=0
file:.git/config        core.filemode=false
file:.git/config        core.bare=false
file:.git/config        core.logallrefupdates=true
file:.git/config        core.symlinks=false
file:.git/config        core.ignorecase=true
```

So, still, `git diff` returns text

```
warning: CRLF will be replaced by LF in dummy.pdf.
The file will have its original line endings in your working directory
diff --git a/dummy.pdf b/dummy.pdf
index 774c2ea..c0e31a0 100644
--- a/dummy.pdf
+++ b/dummy.pdf
@@ -1,2 +1,22 @@
-Dummy PDF file
+A Simple PDF File
+
+  This is a small demonstration .pdf file -
```
and in line the that, the issue does appear.

In summary, I believe VS Code's resource resolver is confused by `git`s tendency to convert resources into text. This is very similar to #129505 - however, this issue appears even with default settings, I believe (on Windows). VS Code should make sure that in any kind of diff view, where the right side of the diff view resolves via the file system and will always receive a binary file, the left side should resolve without resorting to any kind of `textconv`.
